### PR TITLE
Add routing

### PR DESCRIPTION
**Pull request by bazelajoanna**
Add React Router v6 using documentation: 
https://reactrouter.com/docs/en/v6/getting-started/installation

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,14 +1,25 @@
...
     <>
       <Navbar />
-      <ProblemIndex />
+      <Routes>
+        {/* <Route path="/" element {<LandingPage/>}> */}
+        <Route path="/" element={<ProblemIndex />} />
+        <Route path="/problems/:problemId" element={<DetailedProblemView />} />
+        <Route path="/about" element={<AboutMethod />} />
+        <Route path="/moderator" element={<Moderator />} />
+      </Routes>
```
Comment (franiakat) [inline src/App.tsx]: 👏

File: src/assets/styles/global-styles.js [hunk 2]
```diff
@@ -59,8 +59,14 @@ ${fonts}
   a {
     text-decoration: none;
     transition: color 0.05s ease-in-out;
+
```
Comment (franiakat) [inline src/assets/styles/global-styles.js]: what about not visited link colour? :)

File: src/components/molecules/AboutMethod/index.tsx [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import { FC } from "react";
+import { Link } from "react-router-dom";
+
+const AboutMethod: FC = () => {
```
Comment (franiakat) [inline src/components/molecules/AboutMethod/index.tsx]: How about we have a convention that we keep those views/pages in `pages/` directory? We could also agree on a naming convention, e.g.:
`AboutMethodView` or `AboutMethodPage` and align all page-like components also `ProblemIndexPage` or accordingly `ProblemIndexView`.
Btw I would rename it to `ProblemsView` or `ProblemsPage`.

File: src/components/molecules/SearchResultTile/index.tsx [hunk 4]
```diff
@@ -27,9 +31,11 @@ const SearchResultTile: FC<ProblemSummaryProps> = ({
...
         </S.TagWrapper>
       </S.DetailsWrapper>
-      <Button isPrimary={false} isHidden>
-        Zobacz treść
-      </Button>
+      <Link to={`/problems/${id}`}>
+        <Button isPrimary={false} isHidden>
+          Zobacz treść
+        </Button>
+      </Link>
```
Comment (franiakat) [inline src/components/molecules/SearchResultTile/index.tsx]: I think ideal would be that we have Link from react router that is styled as component, as having button tag inside of a link tag seems a bit odd. ;)

File: src/components/organisms/Navbar/index.tsx [hunk 5]
```diff
@@ -48,8 +51,12 @@ const Navbar: FC = () => {
...
       <S.NavbarMenu className={navbarMenuStyle}>
         <S.NavbarMenuList>
-          <S.NavbarMenuItem>O metodzie</S.NavbarMenuItem>
-          <S.NavbarMenuItem>Strefa moderatora</S.NavbarMenuItem>
+          <Link to="/about">
+            <S.NavbarMenuItem>O metodzie</S.NavbarMenuItem>
+          </Link>
+          <Link to="/moderator">
+            <S.NavbarMenuItem>Strefa moderatora</S.NavbarMenuItem>
+          </Link>
```
Comment (franiakat) [inline src/components/organisms/Navbar/index.tsx]: It should be the other way around `li` tags should contain `a` tags.

File: src/components/pages/DetailedProblemView/index.tsx [hunk 6]
```diff
@@ -0,0 +1,44 @@
+import { FC, useEffect, useState } from "react";
+import { Link, useParams } from "react-router-dom";
+import { useQuery } from "@apollo/client";
+
+import { GetProblemsQuery, PROBLEMS } from "../ProblemIndex/interface";
+import { ProblemSummaryProps } from "../../molecules/SearchResultTile/interface";
+
+const DetailedProblemView: FC = () => {
```
Comment (franiakat) [inline src/components/pages/DetailedProblemView/index.tsx]: ProblemDetailedView would be more correct in English, I think 🙃

File: src/components/pages/DetailedProblemView/index.tsx [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+  const params = useParams();
+  const id: any = params.problemId;
+
+  const { data, loading } = useQuery<GetProblemsQuery>(PROBLEMS);
```
Comment (franiakat) [inline src/components/pages/DetailedProblemView/index.tsx]: We can rethink it, when implementing this page precisely, but we need for sure additional call to the API to give us more details about the page, but we should treat it as a separate task, so let's not change the approach here and now. ;)

File: src/components/pages/DetailedProblemView/index.tsx [hunk 8]
```diff
@@ -0,0 +1,44 @@
+import { FC, useEffect, useState } from "react";
+import { Link, useParams } from "react-router-dom";
+import { useQuery } from "@apollo/client";
+
+import { GetProblemsQuery, PROBLEMS } from "../ProblemIndex/interface";
+import { ProblemSummaryProps } from "../../molecules/SearchResultTile/interface";
+
+const DetailedProblemView: FC = () => {
+  const params = useParams();
+  const id: any = params.problemId;
```
Comment (franiakat) [inline src/components/pages/DetailedProblemView/index.tsx]: `number`?
```suggestion
  const id: number = params.problemId;
```

File: src/components/pages/DetailedProblemView/index.tsx [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+      const currentProblem = problems[id - 1];
+      setProblems(filteredProblems);
+      setProblemDetails({
+        title: currentProblem.title,
+        author: currentProblem.author,
+      });
+    }
+  }, [data?.problems.length, loading]);
+
+  return data?.problems && !loading ? (
```
Comment (franiakat) [inline src/components/pages/DetailedProblemView/index.tsx]: It's used twice, can be extracted to a constant.

**Review by franiakat — CHANGES_REQUESTED** [review]
Awesome job, Asia! Happy to see new possible interactions in our app. 🍾

File: src/assets/styles/global-styles.js (hunk 2 above)
Comment (bazelajoanna) [reply src/assets/styles/global-styles.js]: I've removed `color: inherit` from visited link and moved it to an `a` element. Now it works correctly :)

File: src/components/molecules/AboutMethod/index.tsx (hunk 3 above)
Comment (bazelajoanna) [reply src/components/molecules/AboutMethod/index.tsx]: Good point, I've renamed them to "...`Page`" and moved to `pages/` directory :)

File: src/components/pages/DetailedProblemView/index.tsx (hunk 8 above)
Comment (bazelajoanna) [reply src/components/pages/DetailedProblemView/index.tsx]: At first I tried with `number`, but the error occurred. That's why i set `any`. Now when I set as `number` there is the same message: `Type 'string | undefined' is not assignable to type 'number'.
  Type 'undefined' is not assignable to type 'number'`

File: src/components/pages/DetailedProblemView/index.tsx (hunk 9 above)
Comment (bazelajoanna) [reply src/components/pages/DetailedProblemView/index.tsx]: Do you mean `data?.problems`? Or the whole statement `data?.problems && !loading`?

File: src/components/pages/DetailedProblemView/index.tsx (hunk 8 above)
Comment (franiakat) [reply src/components/pages/DetailedProblemView/index.tsx]: I see, yes params would be strings always, so here I'd go with
```suggestion
  const id: string | undefined = params.problemId;
```

File: src/components/pages/DetailedProblemView/index.tsx (hunk 9 above)
Comment (franiakat) [reply src/components/pages/DetailedProblemView/index.tsx]: I meant `data?.problems && !loading`, so basically you'd have sth along those lines:
`const problemsLoaded: boolean = data?.problems && !loading;`

Comment (bazelajoanna) [reply src/components/pages/DetailedProblemView/index.tsx]: Done, thanks :)

File: src/components/pages/DetailedProblemView/index.tsx (hunk 8 above)
Comment (bazelajoanna) [reply src/components/pages/DetailedProblemView/index.tsx]: I've changed it to: `const id: string = params.problemId || "";`

**Review by franiakat — APPROVED** [review]
Really good work, Asia, I am happy to approve 🙃 👏
sqash

**Pull request merged**